internal/os/disk/api.go

```diff
@@ -0,0 +1,61 @@
+package disk
+
+import (
+	"fmt"
+	"os/exec"
+)
+
+// Implements the OS API calls related to Disk Devices. All code here should be very simple
+// pass-through to the OS APIs or cmdlets. Any logic around the APIs/cmdlet invocation
+// should go in internal/server/filesystem/disk.go so that logic can be easily unit-tested
+// without requiring specific OS environments.
+
+type APIImplementor struct{}
+
+func New() APIImplementor {
+	return APIImplementor{}
+}
+
+func (APIImplementor) IsDiskInitialized(diskID string) (bool, error) {
+	cmd := fmt.Sprintf("Get-Disk -Number %s | Where partitionstyle -eq 'raw'", diskID)
```

Comment:
Shouldn't we be looking for partition style MBR since in the initialization we are using MBR as the partition style ?

Reply:
We want to look for `raw` here. `MBR`/`GPT` would indicate disk is already initialized.